### ISSUE TITLE
Widget stopped working 

### ISSUE THREAD

**Issue by Njchris68** [tonesto7/fordpass-scriptable#111]
Widget stopped working tried a new install but no luck.

![3EAD31DE-BF9B-4380-863E-1958A0DDF30A](https://user-images.githubusercontent.com/108413367/176456208-6afbd867-ce93-4272-b71a-d1597c488268.png)

**Comment by MBCook** [comment]
I’ve noticed this as well. It seems it hasn’t been able to actually update the data from the vehicle for two days.

**Comment by MayorVaughn** [comment]
I am also experiencing this behavior. Followed same steps as you. No data update for a few days, tried to delete and reinstall, stuck in the menu loop now. Has anyone found a solution?

**Comment by shadowmite** [comment]
Agreed, my widget has completely stopped functioning in the last couple days.

**Comment by christophermp** [comment]
Ford has pulled the plug. The current way to get a Auth token has been changed. 😒 
Let's hope a new way is found soon.

**Comment by MBCook** [comment]
Shoot I was afraid that was the case.

**Comment by kevcody** [comment]
FWIW, this also affected the Home Assistant Fordpass app… they updated it yesterday and data is flowing again :) 

https://github.com/itchannel/fordpass-ha/tree/1.34

<img width="739" alt="image" src="https://user-images.githubusercontent.com/40125621/176793886-040bf9c3-e9ca-4353-898a-b2e1c488cc19.png">

**Comment by kevcody** [comment]
@tonesto7 here is the OAUTH flow that seems to be working for the Home Assistant Ford Pass app >> https://github.com/itchannel/fordpass-ha/blob/1.34/custom_components/fordpass/config_flow.py

**Comment by kayyovx** [comment]
Having the same issue, slightly over 6 days it's stopped working for me.

**Comment by renho17** [comment]
Same for me. When’s selecting widget color it errors out.

**Comment by tonesto7** [comment]
Ford locked down the authentication side of things. I have a fix coming as soon as I can finish working on it.

**Comment by tonesto7** [comment]
> @tonesto7 here is the OAUTH flow that seems to be working for the Home Assistant Ford Pass app >> https://github.com/itchannel/fordpass-ha/blob/1.34/custom_components/fordpass/config_flow.py

Thanks. I helped to figure it out with those guys. 
The issue is the limitations in the Scriptable JavaScript environment.

**Comment by renho17** [comment]
Ok thanks bro 

                   Ren Reynolds
                      IT Support
                   707-541-8003
               

On Jul 3, 2022, at 10:46 AM, tonesto7 ***@***.***> wrote:

﻿
Ford locked down the authentication side of things. I have a fix coming as soon as I can finish working on it.

**Comment by Berg0987** [comment]
Great work on this widget.  Heads up - you may want to add some code that limits how many calls the script makes to the FordPass API (something gets caught in a loop and makes excessive calls) I have seen other great projects blocked when they see too many calls in a very short period of time- sets off their scanners.

**Comment by banolt** [comment]
Yea man, great job on this widget. Now that it doesn’t work I realise how often i use it . Keep up the amazing work !

**Comment by renho17** [comment]
> > @tonesto7 here is the OAUTH flow that seems to be working for the Home Assistant Ford Pass app >> https://github.com/itchannel/fordpass-ha/blob/1.34/custom_components/fordpass/config_flow.py
> 
> Thanks. I helped to figure it out with those guys. The issue is the limitations in the Scriptable JavaScript environment.

So does this mean it is fixed now?

**Comment by christophermp** [comment]
> > > @tonesto7 here is the OAUTH flow that seems to be working for the Home Assistant Ford Pass app >> https://github.com/itchannel/fordpass-ha/blob/1.34/custom_components/fordpass/config_flow.py
> 
> > 
> 
> > Thanks. I helped to figure it out with those guys. The issue is the limitations in the Scriptable JavaScript environment.
> 
> 
> 
> So does this mean it is fixed now?

Yea, it works again now.
